File: app/src/main/java/com/stephenmorgandevelopment/thelinuxmanual/models/TextSearchResult.java
package com.stephenmorgandevelopment.thelinuxmanual.models;

import java.util.List;
import java.util.Map;

public class TextSearchResult {
	private final String query;
	private final List<SingleTextMatch> matchIndexes;

	public TextSearchResult(String query, List<SingleTextMatch> matchIndexes, int count) {
		this.query = query;
		this.matchIndexes = matchIndexes;
	}

	public String getQuery() {
		return query;
	}

	public int getCount() {
		return matchIndexes.size();
	}

	public List<SingleTextMatch> getMatchIndexes() {
		return matchIndexes;
	}
}

File: app/src/main/java/com/stephenmorgandevelopment/thelinuxmanual/models/TextSearchResult.kt
package com.stephenmorgandevelopment.thelinuxmanual.models

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class TextSearchResult(
    val query: String,
    val textMatches: List<SingleTextMatch>,
) : Parcelable {
    val count get(): Int = textMatches.size

    /**
     * @return SingleTextMatch - text match at index, index 0
     * if out of bounds, or null if list is empty.
     */
    fun getMatch(index: Int): SingleTextMatch? = try {
        textMatches[index]
    } catch (e: IndexOutOfBoundsException) {
        if (textMatches.isNotEmpty()) textMatches[0]
        else null
    }

    fun getSectionAt(index: Int): String? = getMatch(index)?.section
}